src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -4868,7 +4868,7 @@ void C2_MacroAssembler::vector_cast_float_to_int_special_cases_avx(XMMRegister d
   // values are set.
   vpxor(xtmp3, xtmp2, xtmp4, vec_enc);
 
-  vblendvps(dst, dst, xtmp1, xtmp3, vec_enc);
+  vblendvps(dst, dst, xtmp1, xtmp3, vec_enc, true, xtmp4);
```

Comment:
The vblendvps at line 4861 could also be emulated:
From:
  vpxor(xtmp4, xtmp4, xtmp4, vec_enc);
  vcmpps(xtmp3, src, src, Assembler::UNORD_Q, vec_enc);
  vblendvps(dst, dst, xtmp4, xtmp3, vec_enc);

To:
  vpxor(xtmp4, xtmp4, xtmp4, vec_enc);
  vcmpps(xtmp3, src, src, Assembler::UNORD_Q, vec_enc);
  vblendvps(dst, dst, xtmp4, xtmp3, vec_enc, false, xtmp4);

Reply:
Cannot use `xtmp4`, scratch isn't available, (`scratch != src2`):
```
bool scratch_available = scratch != xnoreg && scratch != src1 && scratch != src2 && scratch != dst;
```

(Originally that's exactly what I had, but got caught when I intentionally made the fall-back/default case `assert`)